src/app.js

```diff
@@ -0,0 +1,389 @@
+
+  for (let row = 0; row < GRID_SIZE; row++) {
+    for (let col = 0; col < GRID_SIZE; col++) {
+      const cellValue = gameGrid[row][col];
+      const cellElement = gameField.rows[row].cells[col];
+
+      cellElement.className = 'field-cell';
+
+      if (cellValue !== 0) {
+        cellElement.textContent = cellValue;
+        cellElement.classList.add(`field-cell--${cellValue}`);
+      } else {
+        cellElement.textContent = '';
+      }
+    }
+  }
+}
+
+function initGame() {
+  gameGrid = Array.from({ length: GRID_SIZE }, () => Array(GRID_SIZE).fill(0));
+
+  score = 0;
+  gameIsOver = false;
+
+  generateRandomNumber();
+  generateRandomNumber();
+
+  renderGame();
+
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+
+  startMessage.classList.remove('hidden');
+  startButton.classList.remove('hidden');
+  document.getElementById('restartButton').classList.add('hidden');
+}
+
+function restartGame() {
+  score = 0;
+  gameIsOver = false;
+
+  for (let row = 0; row < GRID_SIZE; row++) {
+    for (let col = 0; col < GRID_SIZE; col++) {
+      gameGrid[row][col] = 0;
+    }
+  }
+
+  generateRandomNumber();
+  generateRandomNumber();
+
+  renderGame();
+
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+
+  startMessage.classList.add('hidden');
+  startButton.classList.add('hidden');
+
+  document.getElementById('restartButton').classList.remove('hidden');
+  document.querySelector('.game-score').textContent = score;
+
+  restartButton.blur();
+}
+
+restartButton.addEventListener('click', () => {
+  restartGame();
+});
+
+function startGame() {
+  hideStartMessage();
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+
+  startButton.classList.add('hidden');
+  document.getElementById('restartButton').classList.remove('hidden');
+}
+
+function hideStartMessage() {
+  startMessage.classList.add('hidden');
+}
+
+function handleKeyPress(move) {
+  if (gameIsOver) {
+    return;
+  }
+
+  const key = move.key;
+
+  if (key === 'ArrowUp'
+    || key === 'ArrowDown' || key === 'ArrowLeft' || key === 'ArrowRight') {
+    hideStartMessage();
+    startButton.classList.add('hidden');
+    restartButton.classList.remove('hidden');
+  }
+
+  switch (key) {
+    case 'ArrowUp':
+      moveUp();
+      break;
+    case 'ArrowDown':
+      moveDown();
+      break;
+    case 'ArrowLeft':
+      moveLeft();
+      break;
+    case 'ArrowRight':
+      moveRight();
+      break;
+    default:
+      break;
+  }
+}
+
+document.addEventListener('keydown', handleKeyPress);
+startButton.addEventListener('click', startGame);
+
+initGame();
+
+function combineCells(row, col, nextRow, nextCol) {
+  const currentValue = gameGrid[row][col];
+  const nextValue = gameGrid[nextRow][nextCol];
+
+  if (currentValue === nextValue) {
+    gameGrid[row][col] = currentValue * 2;
+    gameGrid[nextRow][nextCol] = 0;
+    score += currentValue * 2;
+  }
+}
+
+function moveCellsUp() {
```

Comment:
avoid such nesting, use helper functions instead of
![image](https://github.com/mate-academy/js_2048_game/assets/70009205/a16ffb3b-e60b-4d04-b2d4-8b066a038048)